parsers/CA_YT.py

```diff
@@ -1,128 +1,119 @@
+    does operations, billing, and generation in some of the off-grid
+    communities.
 
     See schema of the grid at http://www.atcoelectricyukon.com/About-Us/
 
-    Per https://en.wikipedia.org/wiki/Yukon#Municipalities_by_population
-    of total population 35874 (2016 census), 28238 are in municipalities
-    that are connected to the grid - that is 78.7%.
+    Per https://en.wikipedia.org/wiki/Yukon#Municipalities_by_population of
+    total population 35874 (2016 census), 28238 are in municipalities that are
+    connected to the grid - that is 78.7%.
 
-    Off-grid generation is with diesel generators, this is not reported online as of 2017-06-23
-    and is not included in this calculation.
+    Off-grid generation is with diesel generators, this is not reported online
+    as of 2017-06-23 and is not included in this calculation.
 
-    Yukon Energy reports only "hydro" and "thermal" generation.
-    Per http://www.yukonenergy.ca/ask-janet/lng-and-boil-off-gas,
-    in 2016 the thermal generation was about 50% diesel and 50% LNG.
-    But since Yukon Energy doesn't break it down on their website,
-    we return all thermal as "unknown".
+    Yukon Energy reports only "hydro" and "thermal" generation. Per
+    http://www.yukonenergy.ca/ask-janet/lng-and-boil-off-gas, in 2016 the
+    thermal generation was about 50% diesel and 50% LNG. But since Yukon Energy
+    doesn't break it down on their website, we return all thermal as "unknown".
 
     Per https://en.wikipedia.org/wiki/List_of_generating_stations_in_Yukon
-    Yukon Energy operates about 98% of Yukon's hydro capacity, the only exception is
-    the small 1.3 MW Fish Lake dam operated by ATCO/Yukon Electrical.
-    That's small enough to not matter, I think.
+    Yukon Energy operates about 98% of Yukon's hydro capacity, the only
+    exception is the small 1.3 MW Fish Lake dam operated by ATCO/Yukon
+    Electrical. That's small enough to not matter, I think.
 
-    There is also a small 0.81 MW wind farm, its current generation is not available.
+    There is also a small 0.81 MW wind farm, its current generation is not
+    available.
     """
-    if target_datetime:
-        raise NotImplementedError("This parser is not yet able to parse past dates")
 
-    requests_obj = session or Session()
-
-    url = "http://www.yukonenergy.ca/consumption/chart_current.php?chart=current&width=420"
-    response = requests_obj.get(url)
-
-    soup = BeautifulSoup(response.text, "html.parser")
-
-    def find_div_by_class(soup_obj, cls):
-        return soup_obj.find("div", attrs={"class": cls})
-
-    def parse_mw(text):
+    def _parse_mw(text):
+        """
+        Extract the power value from the source's HTML text content. The text
+        is formatted as, e.g., "37.69 MW - hydro".
+        """
         try:
-            return float(text[: text.index("MW")])
+            return float(text[: text.index(" MW")])
```

Comment:
Was the additional space here intentional?

Reply:
I did this intentionally so the string passed to `float` wouldn't always contain an extra space (which just gets ignored), but it's not necessary!